### PR TITLE
Simpler xml parsing

### PR DESCRIPTION
**Pull request by DryRun**
Working version of new master snippet parser enforcing XML tag <=> FM parameter (https://github.com/HCALRunControl/levelOneHCALFM/issues/170). Replaces the previous include method (which used HCALxmlHandler::readFile()) with XInclude. Also adds support for specifying VectorT, and MapT parameters in JSON format, which are parsed with the rcms JsonUtil (for example, [1,2,3] or {"key1":"value1","key2":"value2"}).

### CODE REVIEW

**Comment by DryRun** [comment]
I added support for concatenating pre-existing StringT parameters, rather than overwriting. This is needed for HCAL_CFGSCRIPT. The syntax is, e.g., `<HCAL_CFGSCRIPT conflict="concatenate">`. The default is "overwrite".

**Comment by sethcooper** [comment]
This should close #28, right?

**Comment by kakwok** [comment]
no, #219 aka GrandMaster should.

File: src/rcms/fm/app/level1/HCALxmlHandler.java [hunk 1]
```diff
@@ -555,21 +544,16 @@ public void parseMasterSnippet(String selectedRun, String CfgCVSBasePath, boolea
...
                 }
                 String newString = oldString + parameterValue;
-                logger.info("[HCAL " + functionManager.FMname + "]: Old = " + oldString);
-                logger.info("[HCAL " + functionManager.FMname + "]: New = " + newString);
-
                 functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>(parameterName, new StringT(newString)));
               } else {
                 functionManager.getHCALparameterSet().put(new FunctionManagerParameter<StringT>(parameterName, new StringT(parameterValue)));
               }
-              break;
```
Comment (kakwok) [inline src/rcms/fm/app/level1/HCALxmlHandler.java]: I think this line is needed.

File: src/rcms/fm/app/level1/HCALxmlHandler.java [hunk 2]
```diff
@@ -415,38 +439,203 @@ public String getHCALMasterSnippetTagAttribute(String selectedRun, String CfgCVS
...
+      //}
+      //if (functionManager.getHCALparameterSet().contains("TEST_MAPT_STRINGT")) {
+      //  logger.info("[HCAL " + functionManager.FMname + "] Printing test parameter TEST_MAPT_STRINGT: " + //functionManager.getHCALparameterSet().get("TEST_MAPT_STRINGT").getValue());
+      //}
+      //if (functionManager.getHCALparameterSet().contains("TEST_MAPT_INTEGERT")) {
+      //  logger.info("[HCAL " + functionManager.FMname + "] Printing test parameter TEST_MAPT_INTEGERT: " + //functionManager.getHCALparameterSet().get("TEST_MAPT_INTEGERT").getValue());
+      //}
+      
+    } catch ( DOMException | ParserConfigurationException | SAXException | IOException e) {
+        logger.error("[HCAL " + functionManager.FMname + "]: Got an error when parsing masterSnippet: " + e.getMessage());
```
Comment (kakwok) [inline src/rcms/fm/app/level1/HCALxmlHandler.java]: We should goToError when MasterSnippet is not a well-formed XML

**Comment by jhakala** [comment]
Are there plans for this PR?

**Pull request closed without merging**